packages/dx-react-grid/docs/guides/custom-plugin-development.md

```diff
@@ -0,0 +1,709 @@
+# React Grid - Custom Plugin Development
+
+
+## Table of Contents
+1. [Introduction](#introduction)
+2. [Render Custom Markup](#render-custom-markup)
+3. [Share Values Between Plugins](#share-values-between-plugins)
+4. [Use the Functionality of Built-In Plugins](#use-the-functionality-of-built-in-plugins)
+5. [Final Steps](#final-steps)
+
+
+## Introduction
+
+This tutorial is aimed at advanced users who are familiar with [basic Grid concepts](./getting-started.md) and who want to add new functionality to the Grid by using the [React Core](../../../core/docs/guides/fundamentals.md) components.
+
+In this tutorial, we add to the Grid's toolbar a select box that filters the grid by a predefined column. The purpose of the tutorial is not to implement a valuable feature but to show how to extend the Grid's functionality.
+
+We start with the following code. As you can see, we use the Grid in the `Material-UI` theme.
+
+*index.js:*
+```jsx
+import {
+  Grid,
+  Table,
+  TableHeaderRow,
+  Toolbar
+} from '@devexpress/dx-react-grid-material-ui';
+import customers from "./data.js";
+
+class App extends PureComponent {
+  constructor(props) {
+    super(props);
+    this.state = {
+      columns: [
+        { name: 'ID', title: 'ID' },
+        { name: 'CompanyName', title: 'Company Name' },
+        { name: 'Address', title: 'Address' },
+        { name: 'City', title: 'City' },
+        { name: 'State', title: 'State' },
+        { name: 'Zipcode', title: 'Zip code' }
+      ],
+      rows: customers
+    };
+  }
+  render() {
+    const { rows, columns } = this.state;
+    return (
+      <Paper>
+        <Grid
+          rows={rows}
+          columns={columns}
+        >
+          <Table />
+          <TableHeaderRow />
+          <Toolbar />
+        </Grid>
+      </Paper>
+    );
+  }
+}
+```
+*data.js:*
+```js
+export default [{
+    ID: 1,
+    CompanyName: 'Super Mart of the West',
+    Address: '702 SW 8th Street',
+    City: 'Bentonville',
+    State: 'Arkansas',
+    Zipcode: 72716
+  }, {
+    ID: 2,
+    CompanyName: 'Electronics Depot',
+    Address: '2455 Paces Ferry Road NW',
+    City: 'Atlanta',
+    State: 'Georgia',
+    Zipcode: 30339
+  }, {
+    ID: 3,
+    CompanyName: 'K&S Music',
+    Address: '1000 Nicllet Mall',
+    City: 'Minneapolis',
+    State: 'Minnesota',
+    Zipcode: 55403
+  }, {
+    ID: 4,
+    CompanyName: "Tom's Club",
+    Address: '999 Lake Drive',
+    City: 'Issaquah',
+    State: 'Washington',
+    Zipcode: 98027
+  }, {
+    ID: 5,
+    CompanyName: 'E-Mart',
+    Address: '3333 Beverly Rd',
+    City: 'Hoffman Estates',
+    State: 'Illinois',
+    Zipcode: 60179
+  }];
+```
+
+Take a look at the image below to see what you will get by the end of the tutorial. At the top right corner, you can see a select box. When a user selects a value from it, the grid displays only the filtered rows. When the value is cleared using the `CLEAR` button, the grid displays all rows.
```

Comment:
The image below shows the final result. There is a select box at the top right corner. When a user selects a value from it, the grid displays only the filtered rows. When the value is cleared using the `CLEAR` button, the grid displays all rows.

Reply:
Why do you need to use future tense?